pixeltable/exec/exec_context.py

```diff
@@ -1,13 +1,27 @@
-from typing import Optional
+from typing import TYPE_CHECKING, Optional
 
 import sqlalchemy as sql
 
 from pixeltable import exprs
 
+if TYPE_CHECKING:
+    from pixeltable.plan import SampleClause
+
 
 class ExecContext:
     """Class for execution runtime constants"""
 
+    row_builder: exprs.RowBuilder
+    profile: exprs.ExecProfile
+    show_pbar: bool
+    batch_size: int
+    num_rows: Optional[int]
+    conn: Optional[sql.engine.Connection]
+    pk_clause: Optional[list[sql.ClauseElement]]
+    num_computed_exprs: int
+    ignore_errors: bool
+    sample_clause: Optional['SampleClause']
```

Comment:
I meant: put the actual seed into ExecContext (not SampleClause, which is a syntactic element), then refer to that in SampleClause._create_stmt() (which has access to the context).

It doesn't matter at the moment, because we re-generate the entire plan every time we run a query. But the goal for our serving setup is to cache plans and simply initialize new ExecContexts.